src/BizHawk.Client.EmuHawk/CustomControls/HexTextBox.cs

```diff
@@ -135,6 +135,21 @@ protected override void OnTextChanged(EventArgs e)
 			base.OnTextChanged(e);
 		}
 
+		protected override void OnPaste(PasteEventArgs e)
+		{
+			if (e.ContainsText)
+			{
+				string text = e.Text.Trim().RemovePrefix("0x").RemovePrefix('$');
+				if (text.IsHex())
+				{
+					PasteWithMaxLength(text);
+				}
+				e.Handled = true;

```

Comment:
I don't think you're using `Handled` correctly...

Reply:
In what way?

If the text is a valid hex string, manually paste the cleaned string. If it's not hex, it shouldn't be pasted at all. In either case, the default paste handling is suppressed by setting `Handled = true`.

Reply:
On second thought I think that's fine. I'm just used to a `Handled` being used for `event`s, but this is simpler.